packages/aws-cdk/lib/api/cloudwatch-logs.ts

```diff
@@ -0,0 +1,94 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { Mode, SdkProvider } from './aws-auth';
+import { EvaluateCloudFormationTemplate, LazyListStackResources } from './evaluate-cloudformation-template';
+import { CloudWatchLogEventMonitor } from './monitor/logs-monitor';
+
+// resource types that have associated CloudWatch Log Groups that should
+// _not_ be monitored
+const excludedTypes: string[] = ['AWS::EC2::FlowLog', 'AWS::CloudTrail::Trail'];
+
+// resource types that will create a cloudwatch log group
+// with a specific name if one is not provided
+const implicitTypes: string[] = ['AWS::Lambda::Function', 'AWS::CodeBuild::Project'];
+
+export async function registerCloudWatchLogGroups(
+  sdkProvider: SdkProvider,
+  stackArtifact: cxapi.CloudFormationStackArtifact,
+  cloudWatchLogMonitor: CloudWatchLogEventMonitor,
+): Promise<void> {
+
+  const logGroupNames = new Array<string>();
+  const logGroupLogicalIds = new Set<string>();
+  const implicitMap = new Map<string, string>();
+  //
+  // resolve the environment, so we can substitute things like AWS::Region in CFN expressions
+  const resolvedEnv = await sdkProvider.resolveEnvironment(stackArtifact.environment);
+  // create a new SDK using the CLI credentials, because the default one will not work for new-style synthesis -
+  // it assumes the bootstrap deploy Role, which doesn't have permissions to update Lambda functions
+  const sdk = await sdkProvider.forEnvironment(resolvedEnv, Mode.ForReading);
+
+  // add the sdk to the cloudWatchLogMonitor so that it also has the
+  // correct credentials to monitor CloudWatch log groups
+  cloudWatchLogMonitor.addSdk(sdk);
```

Comment:
Probably `setSdk` is the correct name here - it's not like we maintain a collections of `sdk`s, we just overwrite the previous one.

Reply:
Actually, I just realized this will be a problem, right? If I'm using `cdk watch --no-hotswap StackInAccA StackInAccB` with new-style synthesis, this will not work.

I agree it's a little bit of an edge case, so I would be fine shipping this feature with that broken, but at least let's think if it's possible to handle it correctly.

Reply:
I think we may be able to accomplish this by using the lookup role.

Reply:
@skinny85 I refactored this to use an SDK per account. I'm also now using the lookupRole (if using newStyleStackSynthesis) so it should handle the case you mentioned above.